modules/terraform-cdp-deploy/modules/azure/main.tf

```diff
@@ -12,80 +12,127 @@
-
-    plat__azure_idbroker_identity_id      = var.idbroker_identity_id
-    plat__azure_datalakeadmin_identity_id = var.datalakeadmin_identity_id
-    plat__azure_ranger_audit_identity_id  = var.ranger_audit_identity_id
-    plat__azure_log_identity_id           = var.log_identity_id
-    plat__azure_raz_identity_id           = var.raz_identity_id
-
-    }
-  )
-  filename = "cdp_config.yml"
+# ------- CDP Credential -------
+resource "cdp_environments_azure_credential" "cdp_cred" {
+  credential_name = var.cdp_xacccount_credential_name
+  subscription_id = var.subscription_id
+  tenant_id       = var.tenant_id
+  app_based = {
+    application_id = var.xaccount_app_uuid
+    secret_key     = var.xaccount_app_pword
+  }
+  description = "Azure Cross Account Credential for Azure env ${var.environment_name}"
 }
 
-# ------- Create CDP Deployment -------
-resource "null_resource" "cdp_deployment" {
+# ------- CDP Environment -------
+resource "cdp_environments_azure_environment" "cdp_env" {
+  environment_name = var.environment_name
+  credential_name  = cdp_environments_azure_credential.cdp_cred.credential_name
+  region           = var.region
+
+  security_access = {
+    default_security_group_id  = var.security_group_default_uri
+    security_group_id_for_knox = var.security_group_knox_uri
+  }
 
-  # Setup of CDP environment using playbook_setup_cdp.yml Ansible Playbook
-  provisioner "local-exec" {
-    command = "ansible-playbook -vvv -e '@cdp_config.yml' ${path.module}/playbook_setup_cdp.yml"
+  log_storage = {
+    storage_location_base        = var.log_storage_location
+    backup_storage_location_base = var.backup_storage_location
+    managed_identity             = var.log_identity_id
   }
 
-  # Deletion of CDP environment using playbook_teardown_cdp.yml Ansible Playbook
-  provisioner "local-exec" {
-    when    = destroy
-    command = "ansible-playbook -vvv -e '@cdp_config.yml' ${path.module}/playbook_teardown_cdp.yml"
+  public_key = var.public_key_text
+
+  use_public_ip = var.use_public_ips
+  existing_network_params = {
+    resource_group_name = var.resource_group_name
+    network_id          = var.vnet_name
+    subnet_ids          = var.cdp_subnet_names
+  }
+  # TODO: Test once new cdp api is release
+  # endpoint_access_gateway_scheme     = var.endpoint_access_scheme
+  # endpoint_access_gateway_subnet_ids = var.public_subnet_ids
+
+  # Set this parameter to deploy all resources into a single resource group
+  resource_group_name = var.use_single_resource_group ? var.resource_group_name : null
+
+  freeipa = {
+    instance_count_by_group = var.freeipa_instances
   }
 
+  workload_analytics = var.workload_analytics
+  enable_tunnel      = var.enable_ccm_tunnel
+  # tags               = var.tags # NOTE: Waiting on provider fix
+
+  depends_on = [
+    cdp_environments_azure_credential.cdp_cred
+  ]
+
+}
+
+# ------- CDP Admin Group -------
+# Create group
+resource "cdp_iam_group" "cdp_admin_group" {
```

Comment:
Where does the resource definition come from? We do not yet have a `cdp_iam_group` resource in the provider. Does this resource utilise cdpcli bash commands?

Reply:
CDP provider does have a resource to create the group - see https://registry.terraform.io/providers/cloudera/cdp/latest/docs/resources/iam_group. 
We are missing the ability to assign users and roles to the group via the provider however.